storage/replica.go

```diff
@@ -843,19 +843,19 @@ func (r *Replica) Send(ctx context.Context, ba roachpb.BatchRequest) (*roachpb.B
 	if ba.IsAdmin() {
 		log.Trace(ctx, "admin path")
 		br, pErr = r.addAdminCmd(ctx, ba)
-	} else if ba.IsReadOnly() {
-		log.Trace(ctx, "read-only path")
-		br, pErr = r.addReadOnlyCmd(ctx, ba)
 	} else if ba.IsWrite() {
 		log.Trace(ctx, "read-write path")
 		br, pErr = r.addWriteCmd(ctx, ba, nil)
-	} else if len(ba.Requests) == 0 {
+	} else if len(ba.Requests) > 0 {
```

Comment:
Glad you caught the `IsAdmin` thing because I wanted to ask you about it. Currently `IsAdmin` returns true if any of the requests are tagged as `isAdmin`. Moving that check to the last condition would be nice, but it would mean that a batch which contains both `isAdmin` and `isWrite` requests would now be executed differently. It wasn't clear to me if such a batch was legal, though.